src/ja/pixivkomikku/src/eu/kanade/tachiyomi/extension/ja/pixivkomikku/PixivKomikkuUtil.kt

```diff
@@ -0,0 +1,246 @@
+package eu.kanade.tachiyomi.extension.ja.pixivkomikku
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.util.Base64
+import okhttp3.ResponseBody
+import java.io.ByteArrayOutputStream
+import java.security.MessageDigest
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import java.util.TimeZone
+import kotlin.math.abs
+import kotlin.math.ceil
+import kotlin.math.floor
+
+internal fun randomString(): String {
+    // the average length of key
+    val length = (30..40).random()
+
+    return buildString(length) {
+        val charPool = ('a'..'z') + ('A'..'Z') + (0..9)
+
+        for (i in 0 until length) {
+            append(charPool.random())
+        }
+    }
+}
+
+@OptIn(ExperimentalUnsignedTypes::class)
+internal fun getTimeAndHash(): Pair<String, String> {
+    val formatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.ENGLISH)
```

Comment:
pull out the data format from the function body so it doesn't get recreated everytime